src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,131 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75F;
+    private int size = 0;
+    private MyNode<K,V>[] bucketArray = new MyNode[DEFAULT_INITIAL_CAPACITY];
+
+    class MyNode<K,V> {
+        private int hash;
+        private K key;
+        private V value;
+        private MyNode<K, V> next;
+
+        public MyNode(int hash, K key, V value, MyNode<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+
+        public MyNode() {
+
+        }
+    }
+
+    public MyHashMap() {
+
+    }
 
     @Override
     public void put(K key, V value) {
+        int arrayIndex = arrayIndexCalculation(key);
+        MyNode<K,V> tempNode = new MyNode<>();
+        MyNode<K,V> newNode = new MyNode<>(key != null ? key.hashCode() : 0, key, value, null);
+        tempNode = bucketArray[arrayIndex];
 
+        do {
+            if (tempNode == null) {
+                bucketArray[arrayIndex] = newNode;
+                size++;
+                break;
+            } else if (newNode.key == null && tempNode.key == newNode.key) {
+                tempNode.value = newNode.value;
+                break;
+            } else if (newNode.key == null && tempNode.next == null) {
+                tempNode.next = newNode;
+                size++;
+                break;
+            } else if (tempNode.key != null && tempNode.key.equals(newNode.key)) {
+                tempNode.value = newNode.value;
+                break;
+            } else if (tempNode.next == null) {
+                tempNode.next = newNode;
+                size++;
+                break;
+            }
+            tempNode = tempNode.next;
+        } while (tempNode != null);
+        if (size == bucketArray.length * DEFAULT_LOAD_FACTOR) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        return getRemove(key, true);
+    }
+
+    public void clear() {
+        bucketArray = new MyNode[DEFAULT_INITIAL_CAPACITY];
+    }
+
+    public V remove(K key) {
+        return getRemove(key, false);
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private V getRemove(K key, boolean getTrue) {
```

Comment:
This flag - boolean getTrue - is something unclear to me. I can not understand what it means - getTrue. When should I pass true, when false? What is the difference? 
In general, using boolean flags as a method parameter is a bad practice.
I suggest deleting this method at all and having different logic in `getValue(K key)` and `remove(K key)`

Reply:
Comment in slack